spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -896,6 +921,11 @@ private void commitIfNecessary() {
 			}
 		}
 
+		@Override
+		public void seek(String topic, int partition, long offset) {
+			this.seeks.add(new TopicPartitionInitialOffset(topic, partition, offset));
```

Comment:
`KafkaMessageListenerContainer` has `1-1` relationship with the `ListenerConsumer`. So, I believe, the question is closed here.

The `ConcurrentMessageListenerContainer` builds a set of `containers` based on the `partitionSubset(containerProperties, i)`. Together with the `Collection<TopicPartition> getAssignedPartitions()`, it seems for me we can reach a uniqueness and correctly assign requested `seek` to the appropriate `KafkaMessageListenerContainer`.
No ?